packages/eas-cli/src/analytics.ts

```diff
@@ -0,0 +1,82 @@
+import Segment from 'analytics-node';
+import os from 'os';
+
+/**
+ * We use require() to exclude package.json from TypeScript's analysis since it lives outside
+ * the src directory and would change the directory structure of the emitted files
+ * under the build directory
+ */
+const packageJSON = require('../package.json');
+
+const PLATFORM_TO_ANALYTICS_PLATFORM: { [platform: string]: string } = {
+  darwin: 'Mac',
+  win32: 'Windows',
+  linux: 'Linux',
+};
+
+// values shared between different instances of Segement
+let sharedUsername: string | undefined;
+let sharedUserTraits: any;
```

Comment:
Rather than adding these here, maybe make the `AnalyticsClient` a singleton and have these as instance properties?

Reply:
`AnalyticsClient` can't be a singleton because we need separate instances to track different stuff. EAS Build events will be reported to different Segment source than e.g. EAS submit our updates. Default client exported here might be used for generic cli analytics (not sure if we will need it).

That separation is necessary because processing this data on analytics service to e.g. track user builds across `eas-cli` ->`www` -> `turtle` would be problematic and doing it this way we can just merge few sources and we have only build related data for all services